kernel/cbytegen.ml

```diff
@@ -411,7 +446,7 @@ let comp_app comp_fun comp_arg reloc f args sz cont =
 	 comp_fun reloc f (sz + nargs)
 	   (Kappterm(nargs, k + nargs) :: (discard_dead_code cont)))
   | None ->
-      if nargs < 4 then
+      if nargs <= 4 then
```

Comment:
I've no clue what this is doing, but in the code below I see `+3` (not being changed).
Is category theory wrong here?

Reply:
Indeed, this is on my list of things to check.

Reply:
The `3` below is the size of the stack part corresponding to `Kpush_retaddr`, so it is orthogonal to the number of arguments in the application.